03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
+exists. We are setting the NA value to -9999 for consistency. 
+####CONSISTENCY WITH WHAT?  NOT PREVIOUSLY MENTIONED
 
-```{r SJER-CHM, echo=FALSE }
 
-##load the DTM
-DTM_SJER <- raster("NEON_RemoteSensing/SJER/DTM/SJER_dtmCrop.tif")
-#load the DSM
-DSM_SJER <- raster("NEON_RemoteSensing/SJER/DSM/SJER_dsmCrop.tif")
+#Challenge: Exploring the San Joaquin field site.  
+As ecologists we may want to explore the phenology of two different sites.  The 
+San Joaquin valley in California is rather different from the Harvard Forest in 
+Massachusetts and could make a good comparison.  
 
-#use overlay to subtract the two rasters
-CHM_ov_SJER <- overlay(DSM_SJER,DTM_SJER,fun=subtRasters)
+LiDAR and imagery data used to create the rasters were collected at the San 
+Joaquin NEON field sites the same as for the Harvard Forest NEON field sites. 
 
-#plot the output
-plot(CHM_ov_SJER,
-     main="NEON Canopy Height Model - Overlay Subtract\n SJER")
+Your challenge is to import the San Joaquin raster files so that we can compare 
+to two sites. So that we can use object/file names in future lessons names 
+should follow the format of objectType_SJER (e.g. `DSM_SJER`).
 
-```
+1. Import the DSM and DTM from the SJER directory. Don't forget to examine the
+data for CRS, bad values, etc. 
+2. Create a CHM from the two raster layers and check to make sure the data are 
+what you expect. 
+3. Plot the CHM from SJER. 
+4. Export as a GeoTIFF.
+5. What do you notice about tree heights at the two sites?
 
+```{r SJER-CHM, }
+#1.
+#load the DTM
+DTM_SJER <- raster("NEON_RemoteSensing/SJER/DTM/SJER_dtmCrop.tif")
+#load the DSM
+DSM_SJER <- raster("NEON_RemoteSensing/SJER/DSM/SJER_dsmCrop.tif")
 
-2. Once you are done with the above, plot a histogram of both CHM's. What do you notice
-about tree heights across the two sites?
+#check CRS, units, etc
+DTM_SJER@data
+DSM_SJER@data
 
-```{r histogram-compare, echo=FALSE}
+#check values
+hist(DTM_SJER, maxpixels=ncell(DTM_SJER))
+hist(DSM_SJER, maxpixels=ncell(DSM_SJER))
 
+#2.
+#use overlay to subtract the two rasters
+CHM_ov_SJER <- overlay(DSM_SJER,DTM_SJER,fun=subtRasters)
 
-#view histogram
 hist(CHM_ov_SJER, 
      main="NEON Canopy Height Model - Histogram\n SJER")
 
-#view histogram
-hist(CHM_ov_HARV, 
-     main="NEON Canopy Height Model - Histogram\n HARV")
-
-```
-
-
-```{r notes-chunk }
-
- #Compare with CHM raster, should have different x-axis ranges
+#3
+#plot the output
+plot(CHM_ov_SJER,
+     main="NEON Canopy Height Model - Overlay Subtract\n SJER")
 
-# Challenge: play with resolution (i.e., pixel numbers)
+#4 
+#Write to object to file
+writeRaster(CHM_ov_SJER,"chm_ov_SJER.tiff",
+            format="GTiff", 
+            overwrite=TRUE, 
+            NAflag=-9999)
 
-# TODO: reprojection of a single-band raster, ask others? 
-# TODO: summarizing multiple pixels into one value
-# TODO: do raster math on single raster
+#4.  Tree heights are WAY shorter in SJER. 
```

Comment:
Another cool challenge. We could show them the code for PAR / layout and have them plot the histograms side by side??